src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/NetworkUsageInterceptor.java

```diff
@@ -0,0 +1,48 @@
+package io.stargate.sgv2.jsonapi.service.embedding.operation;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.io.ByteStreams;
+import com.google.common.io.CountingOutputStream;
+import jakarta.ws.rs.client.ClientRequestContext;
+import jakarta.ws.rs.client.ClientResponseContext;
+import jakarta.ws.rs.client.ClientResponseFilter;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.logging.Logger;
+
+public class NetworkUsageInterceptor implements ClientResponseFilter {
+
+  private static final Logger LOGGER = Logger.getLogger(NetworkUsageInterceptor.class.getName());
+  private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper(); // Jackson object mapper
+
+  @Override
+  public void filter(ClientRequestContext requestContext, ClientResponseContext responseContext)
+      throws IOException {
+    int receivedBytes = 0;
+    int sentBytes = 0;
+    if (requestContext.hasEntity()) {
+      try {
+        ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
```

Comment:
Ok, almost what I meant, but ideally there is no "real" output stream at all -- but "null" output stream. While it's easy to write on your own, I think Guava has one as per (https://stackoverflow.com/questions/691813/is-there-a-null-outputstream-in-java):

https://guava.dev/releases/19.0/api/docs/com/google/common/io/ByteStreams.html#nullOutputStream()

This to avoid all memory allocation `ByteArrayOutputStream` does.

Reply:
Resolved also used the same for the response bytes.